### PR TITLE
gh-120619: Optimize through `_Py_FRAME_GENERAL`

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
Previously every trace with _PY_FRAME_GENERAL wasn't getting optimized, which is a huge figure.

* Issue: gh-120619

### CODE REVIEW

File: Python/optimizer_bytecodes.c [hunk 1]
```diff
@@ -619,12 +619,30 @@ dummy_func(void) {
...
-        (void)self_or_null;
-        (void)args;
-        new_frame = NULL;
-        ctx->done = true;
+        (void)(self_or_null);
+        (void)(callable);
+        PyCodeObject *co = NULL;
+        assert((this_instr + 2)->opcode == _PUSH_FRAME);
+        uint64_t push_operand = (this_instr + 2)->operand;
+        if (push_operand & 1) {
```
Comment (markshannon) [inline Python/optimizer_bytecodes.c]: This section looks the same as the code in `_INIT_CALL_PY_EXACT_ARGS`. Could you factor it out?

**Review by markshannon — COMMENTED** [review]
Looks good in general.
Do you want me to run stats on this?

**Comment by Fidget-Spinner** [comment]
No need for stats. _PY_FRAME_GENERAL is pretty low

**Review by markshannon — APPROVED** [review]
Looks good, thanks

**Pull request merged**